### PR TITLE
Docs/Add FAQ

### PR DESCRIPTION
**Pull request by EugeniyBurmistrov**

### CODE REVIEW

File: docs/faq.md [hunk 1]
```diff
@@ -3,3 +3,66 @@ id: faq
 title: FAQ
 ---
 
+**Q**: Where can I learn about CQRS and Event Sourcing?<br/>
```
Comment (dirk-pieterse) [inline docs/faq.md]:
```suggestion
**Q**: Where can I find information about CQRS and Event Sourcing?<br/>
```

File: docs/faq.md [hunk 2]
```diff
@@ -3,3 +3,66 @@ id: faq
...
+**A**: Refer to the following resources:
+
+- [Martin Fowler's Enterprise Architecture pattern](https://martinfowler.com/eaaDev/EventSourcing.html)
+- [Greg Young classic explanation talk](https://www.youtube.com/watch?v=8JKjvY4etTY)
+- [Greg Young's EventStore docs: Event Sourcing Basics](https://eventstore.org/docs/event-sourcing-basics/index.html)
+- [Greg Young's DDD CQRS Class](https://www.youtube.com/watch?v=whCk1Q87_ZI)
+- [Event Sourcing Made Simple](https://kickstarter.engineering/event-sourcing-made-simple-4a2625113224)
+- [Migrating to Microservices and Event-Sourcing: the Dos and Dont’s](https://hackernoon.com/migrating-to-microservices-and-event-sourcing-the-dos-and-donts-195153c7487d)
+- [CQRS.nu](http://www.cqrs.nu)
+- [Event Sourcing: What it is and why it's awesome](https://barryosull.com/blog/event-sourcing-what-it-is-and-why-it-s-awesome), read other related posts at https://barryosull.com/blog
```
Comment (dirk-pieterse) [inline docs/faq.md]:
```suggestion
- [Event Sourcing: What it is and why it's awesome](https://barryosull.com/blog/event-sourcing-what-it-is-and-why-it-s-awesome) and other related posts at https://barryosull.com/blog
```

File: docs/faq.md [hunk 3]
```diff
@@ -3,3 +3,66 @@ id: faq
...
+
+**Q**: What is the difference between a Read Model and a View Model?<br/>
+**A**: Read Models implement the standard event sourcing mechanisms.
+View Models are a Redux-specific extension to these mechanisms. View models are reactive, they use websockets to synchronize their state with the redux state on the client.
```
Comment (dirk-pieterse) [inline docs/faq.md]:
```suggestion
View Models are a Redux-specific extension to these mechanisms. View models are reactive and use websockets to synchronize their state with the redux state on the client.
```

File: docs/faq.md [hunk 4]
```diff
@@ -3,3 +3,66 @@ id: faq
...
+  aggregateName: 'User',
+  aggregateId: event.aggregateId,
+  type: 'requestConfirmUser',
+  payload: event.payload
+})
+```
+
+<!-- prettier-ignore-end -->
+
+**Q**: How to perform uniqueness validation for input values?<br/>
```
Comment (dirk-pieterse) [inline docs/faq.md]: uniqueness validation?

File: docs/faq.md [hunk 5]
```diff
@@ -3,3 +3,66 @@ id: faq
...
+**A**: In a distributed application, it is impossible to reliably perform such checks. Instead, you should detect value duplicates in a Read Model or Saga projection code and mark duplicated values as incorrect.
```
Comment (dirk-pieterse) [inline docs/faq.md]: "impossible to reliably perform such checks" Explain

File: docs/faq.md (hunk 4 above)
Comment (EugeniyBurmistrov) [reply docs/faq.md]: to ensure that input values are unique

File: docs/faq.md (hunk 5 above)
Comment (EugeniyBurmistrov) [reply docs/faq.md]: In a distributed system, you can never be sure that all nodes have all the same data at any moment, so its unreliable to just check input values for uniqueness as they arrive to the node.

File: docs/faq.md (hunk 4 above)
Comment (dirk-pieterse) [reply docs/faq.md]:
```suggestion
**Q**: How to perform validation to ensure input values are unique?<br/>
```

File: docs/faq.md (hunk 5 above)
Comment (dirk-pieterse) [reply docs/faq.md]:
```suggestion
**A**: In a distributed application, it is not possible to perform reliable checks. You should detect value duplicates in a Read Model or Saga projection code and mark duplicated values as incorrect.
```

**Pull request merged**